### PR TITLE
Better parameter checking & verbose output

### PR DESCRIPTION
**Pull request by mathiasscheffe**
The better parameter checking allows to also replace with "". The old way did not allow this.

### CODE REVIEW

File: package.json
Comment (aganglada) [inline package.json:2]: [q] Why are we changing the name of the package? I guess you did have a fork of it.

File: package.json
Comment (aganglada) [inline package.json:3]: Next version should be `1.0.5`

File: package.json
Comment (aganglada) [inline package.json:11]: we should keep this the same, I will add you to all-contributors when merged.

File: package.json
Comment (aganglada) [inline package.json:19]: same here

File: README.md
Comment (aganglada) [inline README.md:109]:
```suggestion
        loader: 'pattern-replace-loader',
```

**Review by aganglada — CHANGES_REQUESTED** [review]
@mathiasscheffe Thank you for adding this improvement to my package.

I believe we should keep the `package.json`, expect for the version change.

File: index.js
Comment (aganglada) [inline index.js:5]: [sug] I believe `options.search && options.replace` will catch more edge cases. WDYT?

File: index.js
Comment (aganglada) [inline index.js]: [sug] It will be better if we avoid nesting `if`s, something like: `if (options && options.verbose)` will better. WDYT?

File: package.json
Comment (mathiasscheffe) [reply package.json:2]: Yes, I have enabled that maintainers can edit the PR. The package.json is only different because I created a fork of it. Just do ignore the change when merging the PR.

File: package.json
Comment (mathiasscheffe) [reply package.json:3]: Yes, I have enabled that maintainers can edit the PR. The package.json is only different because I created a fork of it. Just do ignore the change when merging the PR.

File: package.json
Comment (mathiasscheffe) [reply package.json:11]: Yes, I have enabled that maintainers can edit the PR. The package.json is only different because I created a fork of it. Just do ignore the change when merging the PR.

File: package.json
Comment (mathiasscheffe) [reply package.json:19]: Yes, I have enabled that maintainers can edit the PR. The package.json is only different because I created a fork of it. Just do ignore the change when merging the PR.

File: index.js
Comment (mathiasscheffe) [reply index.js:5]: your suggestion it the original implementation. In the case that option.replace it the empty string "" this won't work. The empty string is counted as false. Therefore I have changed to the above implementation.

File: README.md
Comment (mathiasscheffe) [reply README.md:109]: Yes, I have enabled that maintainers can edit the PR. The readme is different because I created a fork of it. Just do modify the loader name when merging it. For my fork the above loader comment is correct can you please fix it during merge?

File: index.js
Comment (mathiasscheffe) [reply index.js]: fixed

File: index.js
Comment (aganglada) [reply index.js:5]: Good catch, in this case I will suggest `Boolean(options.search)` and `Boolean(options.replace)`. Thanks for making those changes.

File: README.md
Comment (aganglada) [reply README.md:109]: I don't believe so. This will change my original package, therefore we should keep it the same.

File: index.js
Comment (aganglada) [reply index.js:5]: After we merge, I'm going to create a new version with your changes and then you should be able to use that version on your project.

File: README.md
Comment (aganglada) [reply README.md:109]: After we merge, I'm going to create a new version with your changes and then you should be able to use that version on your project.

Comment (mathiasscheffe) [reply README.md:109]: This sounds like a plan :). Feel free to edit the loader name.

File: index.js
Comment (mathiasscheffe) [reply index.js:5]: I executed the following code in the browser console:

`let a = Boolean('');
console.log(a);`

And it outputs "false". This means with your suggested change a replace with empty string is still not possible.

Comment (aganglada) [reply index.js:5]: Ahh okay, thats what you want to do:

In this case let's make it implicit then.

What do you think about:
```
const searchDefined = Boolean(options.search) || options.search === ''
```

this will make sure all edge cases are still in place and add empty string. Sorry about the misunderstanding.

Please change them to `const` as those values aren't going to change.

File: README.md
Comment (aganglada) [reply README.md:109]: Please remove the `package.json` changes expect for the version bump up and make it `1.0.5` so we are able to merge it.

File: index.js
Comment (aganglada) [inline index.js:26]:
```suggestion
    if (options && options.verbose) {
```

In this case the above will be enough, also make sure you keep spaces between `if` and `(` **and** `)` and `{`

File: index.js
Comment (aganglada) [inline index.js:27]: [req] Please use single quotes

File: index.js
Comment (aganglada) [inline index.js:28]: [req] add a new line after this to keep the code properly spaced

**Comment by mathiasscheffe** [comment]
I will start with a fresh pull request to have the package.json correct & the readme unchanged.

**Pull request closed without merging**
